service/controller/resource/drainerinitializer/resource.go

```diff
@@ -262,3 +277,31 @@ func (r *Resource) privateDNSForInstance(ctx context.Context, instanceID string)
 
 	return privateDNS, nil
 }
+
+func (r *Resource) completeLifeCycleHook(ctx context.Context, instanceID, asgName string) error {
```

Comment:
I guess this is copied from the finalizer. It is weird that the initializer now does finalizer work. But it might be that we need to re-write the whole thing at some point anyway.

Reply:
Yep